hashring/hashring.go

```diff
@@ -129,88 +135,117 @@ func (r *HashRing) AddServer(address string) bool {
 }
 
 // This function isn't thread-safe, only call it when the HashRing is locked.
-func (r *HashRing) addServerNoLock(address string) bool {
-	if _, ok := r.serverSet[address]; ok {
+func (r *HashRing) addServerNoLock(server membership.Member) bool {
+	if _, ok := r.serverSet[server.GetAddress()]; ok {
```

Comment:
I see what you are getting at. I think it is safe to inline the `addReplicasNoLock` in here which will make the whole function more coherent.